babel.go

```diff
@@ -15,30 +15,29 @@ const DefaultPoolSize = 1
 type babelTransformer struct {
 	Runtime   *goja.Runtime
 	Transform func(string, map[string]interface{}) (goja.Value, error)
-	Ready     chan struct{}
 }
 
 func (t *babelTransformer) Done() {
-	t.Ready <- struct{}{}
+	globalpool <- t
 }
 
-var mux sync.RWMutex
-var pool []*babelTransformer
+var once = &sync.Once{}
+var globalpool chan *babelTransformer
 
-func Init(poolSize int) error {
-	mux.Lock()
-	defer mux.Unlock()
-	pool = make([]*babelTransformer, poolSize)
-	for i := 0; i < poolSize; i++ {
-		vm := goja.New()
-		transformFn, err := loadBabel(vm)
-		if err != nil {
-			return err
+func Init(poolSize int) (err error) {
+	once.Do(func() {
+		globalpool = make(chan *babelTransformer, poolSize)
+		for i := 0; i < poolSize; i++ {
+			vm := goja.New()
+			transformFn, e := loadBabel(vm)
+			if e != nil {
+				err = e
```

Comment:
What do you think about returning the error immediately rather than continuing to populate the pool?

```suggestion
				err = e
				return
```

Reply:
That is a great suggestion!